buildenv/jenkins/common/build

```diff
@@ -308,7 +308,7 @@ def archive() {
                     def server = Artifactory.server ARTIFACTORY_SERVER
                     // set connection timeout to 10 mins to avoid timeout on slow platforms
                     server.connection.timeout = 600
-                    def artifactory_upload_dir = "${ARTIFACTORY_REPO}/${JOB_NAME}/${BUILD_ID}/"
+                    def artifactory_upload_dir = "${ARTIFACTORY_REPO}/${JENKINS_DOMAIN}${JOB_NAME}/${BUILD_ID}/"
```

Comment:
Why are we adding an additional directory layer?  Assuming this is build on #4705, could you not just override artifactory_repo with a different value?